openedx/adg/lms/applications/tests/test_views.py

```diff
@@ -0,0 +1,446 @@
+"""
+Tests for all the views in applications app.
+"""
+
+from datetime import date
+
+import mock
+import pytest
+from django.test import RequestFactory, Client
+from django.urls import reverse
+
+from openedx.adg.lms.applications.models import ApplicationHub
+from openedx.adg.lms.applications.views import ApplicationHubView, ApplicationSuccessView
+from student.tests.factories import UserFactory
+from .constants import USERNAME, EMAIL, PASSWORD
+
+
+@pytest.mark.django_db
+@pytest.fixture()
+def user_for_testing(db):
+    """
+    Create a test user and their corresponding ApplicationHub object
+
+    Returns:
+        User object
+
+    """
+    user = UserFactory(username=USERNAME, email=EMAIL, password=PASSWORD)
+    ApplicationHub.objects.create(user=user)
+    return user
+
+
+@pytest.fixture(scope='module')
+def request_factory():
+    """
+    Returns the request factory to make requests.
+
+    Returns:
+         RequestFactory object
+
+    """
+    return RequestFactory()
+
+
+@pytest.fixture()
+def get_request_for_application_hub_view(request_factory, user_for_testing):
+    """
+    Return a HttpRequest object for application hub get
+
+        Args:
+            request_factory: RequestFactory object
+            user_for_testing: User object
+
+        Returns:
+            HttpRequest object
+
+    """
+    request = request_factory.get(reverse('application_hub'))
+    request.user = user_for_testing
+    return request
+
+
+@pytest.fixture()
+def logged_in_client(user_for_testing):
+    """
+    Return a logged in client
+
+    Args:
+        user_for_testing: User object
+
+    Returns:
+        Client() object with logged in user
+
+    """
+    client = Client()
+    client.login(username=user_for_testing.username, password=PASSWORD)
+    return client
+
+
+@pytest.mark.django_db
+def test_get_redirects_without_login_for_application_hub_view():
+    """
+    Test the case where an unauthenticated user is redirected to login page or not.
+
+    Returns:
+        Boolean
+
+    """
+    response = Client().get(reverse('application_hub'))
+    assert '/login?next=/application/' in response.url
+
+
+@pytest.mark.django_db
+def test_post_user_redirects_without_login_for_application_hub_view():
+    """
+    Test the case where an unauthenticated user is redirected to login page or not.
+
+    Returns:
+        Boolean
+
+    """
+    response = Client().post(reverse('application_hub'))
+    assert '/login?next=/application/' in response.url
+
+
+@pytest.mark.django_db
+@mock.patch('openedx.adg.lms.applications.views.render')
+def test_get_initial_application_state_for_application_hub_view(mock_render, user_for_testing,
+                                                                get_request_for_application_hub_view):
```

Comment:
We do have user in **get_request_for_application_hub_view** request as you have added above in the util method.